src/main/java/org/apache/pulsar/manager/PulsarApplicationListener.java

```diff
@@ -0,0 +1,90 @@
+import org.apache.pulsar.manager.entity.EnvironmentsRepository;
+import org.apache.pulsar.manager.utils.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * PulsarApplicationListener do something after the spring framework initialization is complete.
+ */
+@Component
+public class PulsarApplicationListener implements ApplicationListener<ContextRefreshedEvent> {
+
+    private static final Logger log = LoggerFactory.getLogger(PulsarApplicationListener.class);
+
+    private final EnvironmentsRepository environmentsRepository;
+
+    @Value("${environment.default.name}")
+    private String environmentDefaultName;
+
+    @Value("${environment.default.broker}")
+    private String environmentDefaultBroker;
+
+    @Value("${backend.jwt.token}")
+    private String pulsarJwtToken;
+
+    public PulsarApplicationListener(EnvironmentsRepository environmentsRepository) {
+        this.environmentsRepository = environmentsRepository;
+    }
+
+    @Override
+    public void onApplicationEvent(ContextRefreshedEvent event) {
+        log.info("Start onApplicationEvent");
+        Page<EnvironmentEntity> environmentEntities = environmentsRepository
+                .getEnvironmentsList(1, 1);
+        if (environmentEntities.getResult().size() <= 0) {
+            Optional<EnvironmentEntity> environmentEntityOptional = environmentsRepository
+                    .findByName(environmentDefaultName);
+            if (environmentDefaultName != null
+                    && environmentDefaultBroker != null
+                    && environmentDefaultName.length() > 0
+                    && environmentDefaultBroker.length() > 0
+                    && !environmentEntityOptional.isPresent()) {
+                Map<String, String> header = Maps.newHashMap();
+                header.put("Content-Type", "application/json");
+                if (StringUtils.isNotBlank(pulsarJwtToken)) {
+                    header.put("Authorization", String.format("Bearer %s", pulsarJwtToken));
+                }
+                String httpTestResult = HttpUtil.doGet(environmentDefaultBroker + "/metrics", header);
+                if (httpTestResult != null) {
+                    EnvironmentEntity environmentEntity = new EnvironmentEntity();
+                    environmentEntity.setBroker(environmentDefaultBroker);
+                    environmentEntity.setName(environmentDefaultName);
+                    environmentsRepository.save(environmentEntity);
+                    log.info("The default environment set successful");
+                } else {
+                    log.error("Please check if the network is not working");
```

Comment:
We should fail the boot, otherwise the init sequence is not guaranteed to compete as expected by the user

Reply:
@eolivelli Thank you for your review. I have fixed it, PTAL